x-pack/plugins/infra/server/lib/alerting/common/utils.ts

```diff
@@ -134,3 +141,64 @@ export const getAlertDetailsUrl = (
   spaceId: string,
   alertUuid: string | null
 ) => addSpaceIdToPath(basePath.publicBaseUrl, spaceId, `/app/observability/alerts/${alertUuid}`);
+
+export const NUMBER_OF_DOCUMENTS = 10;
+export const KUBERNETES_POD_UID = 'kubernetes.pod.uid';
+export const termsAggField: Record<string, string> = { [KUBERNETES_POD_UID]: 'container.id' };
+
+export interface AdditionalContext {
+  [x: string]: any;
+}
+
+export const doFieldsExist = async (
+  esClient: ElasticsearchClient,
+  fields: string[],
+  index: string
+): Promise<Record<string, boolean>> => {
+  // Get all supported fields
+  const respMapping = await esClient.fieldCaps({
+    index,
+    fields: '*',
+  });
+
+  const fieldsExisted: Record<string, boolean> = {};
+  const acceptableFields: Set<string> = new Set();
+
+  Object.entries(respMapping.fields).forEach(([key, value]) => {
+    const fieldTypes = Object.keys(value) as ES_FIELD_TYPES[];
+    const isSupportedType = fieldTypes.some((type) => SUPPORTED_ES_FIELD_TYPES.includes(type));
+
+    // Check if fieldName is something we can aggregate on
+    if (isSupportedType) {
+      acceptableFields.add(key);
+    }
+  });
+
+  fields.forEach((field) => {
+    fieldsExisted[field] = acceptableFields.has(field);
+  });
+
+  return fieldsExisted;
+};
+
+export const validGroupByForContext = [
+  'host.name',
+  'host.hostname',
+  'host.id',
+  'kubernetes.pod.uid',
+  'container.id',
+];
+
+export const hasAdditionalContext = (groupBy: string | string[] | undefined) => {
```

Comment:
Nit: I would suggest making this function pure and passing `validGroupByForContext` as a function argument